### PR TITLE
Issue #223 for Documentation Generation Created Github Actions Workflow

### PR DESCRIPTION
**Pull request by UniversalDevGenius**
Issue #223 for Documentation Generation Created Github Actions Workflow , this include adding GenDocs.yml
NOTE: this feature needed `gh-pages` branch need to be created as a prerequisite .
## Types of changes

What types of changes does your code introduce to the community Python SDK?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update or tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I have lint'ed all of my code using repo standards
- [x] I have tested that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: .github/workflows/GenDocs.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        run: echo "MAJOR_VERSION=$(echo $TAG_NAME | cut -d '.' -f 1)" >> "$GITHUB_ENV" # This extracts the major version assuming a 'vX.Y.Z' format
+
+      - name: Generate Documentation for the current tag
+        run: |
+          mkdir -p docs/$MAJOR_VERSION # Note: Changed to use MAJOR_VERSION to create/update the major version folder
+          ROOT_PATH=$(pwd)
+          cd $ROOT_PATH/deepgram
+          pdoc $ROOT_PATH/deepgram -o $ROOT_PATH/docs/$MAJOR_VERSION # Documentation is generated/updated in the major version folder
+
+      - name: Checkout the gh-pages branch
```
Comment (dvonthenen) [inline .github/workflows/GenDocs.yml:47]: Wouldn't you want to change the branch before generating the documentation? This is unlikely to happen but could fail to check out.

File: .github/workflows/GenDocs.yml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+          ref: gh-pages
+          path: gh-pages
+
+      - name: Prepare gh-pages directory structure
+        run: |
+          mkdir -p gh-pages/docs/$MAJOR_VERSION
+      
+      - name: Copy documentation to the gh-pages branch directory
+        run: |
+          cp -r docs/$MAJOR_VERSION gh-pages/docs
```
Comment (dvonthenen) [inline .github/workflows/GenDocs.yml:59]: Why not check out first and generate the pages in the correct path to begin with?

File: .github/workflows/GenDocs.yml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      
+      - name: Deploy to GitHub Pages
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          publish_dir: ./gh-pages
+          publish_branch: gh-pages
+          keep_files: true
```
Comment (dvonthenen) [inline .github/workflows/GenDocs.yml:67]: What does this option do?

**Review by dvonthenen — COMMENTED** [review]
Just some questions for my understanding.

File: .github/workflows/GenDocs.yml (hunk 3 above)
Comment (UniversalDevGenius) [reply .github/workflows/GenDocs.yml:67]: This is the standard GitHub Action that Publish the generated documentation into GitHub Pages.

**Comment by dvonthenen** [comment]
Approved, just waiting for our GH admin to get back from PTO to double check project settings

**Pull request merged**